### DISCUSSION TITLE
EnvVarFixup DLL Not Found

### DISCUSSION THREAD

**Discussion by EmperorLrrr** [TimMangan/MSIX-PackageSupportFramework#25] [General]
Hi
I downloaded the latest version of the MSIX-PackageSupportFramework project and I put the files into the root of my MSIX package.  When I try to run the packaged MSIX program, it doesn't work.  I ran the debug tools and found that PsfRuntime cannot find EnvVarFixup32.dll.   I double checked my package and the EnvVarFixup32.dll is at the location PsfRuntime is looking at (see attached screenshots).  

Any ideas on what I could try?

Thanks,
EmperorLrrr
![EnvVarFixup in package](https://github.com/TimMangan/MSIX-PackageSupportFramework/assets/95937528/9f5008e4-0c98-4fac-9fee-2388e8199db1)
![debug output](https://github.com/TimMangan/MSIX-PackageSupportFramework/assets/95937528/1b0de3cc-9db4-4ab2-b42c-5659595dcfa2)

**Comment by TimMangan** [comment]
EmperorLrrr,

Thank you for reaching out. This looks odd.  The expectation is that it should have been found when it checked at the root folder (a couple lines down from the highlighted one.

It is likely that you did not add the dependency dlls needed by EnVarFixup (as well as RegLegacyFixup).  A procmon trace would have showed an attempt to load the dll from the root folder, followed by the automatic looking for it’s dependencies (which wouldn’t be logged)

It looks like the dependencies weren’t listed for EnvVarFixup (something I’ll address when I get a chance as I’m on the road currently).  But as they are the same as for RegLegacyFixup you can find them here: MSIX-PackageSupportFramework/fixups/RegLegacyFixups at develop · TimMangan/MSIX-PackageSupportFramework · GitHub<https://github.com/TimMangan/MSIX-PackageSupportFramework/tree/develop/fixups/RegLegacyFixups> .

PS: In the screen shot of the files, it looks like the name of the AppXBlockMap changed.  That is very bad but I assume that was done as part of debugging.

Regards,

Tim



From: EmperorLrrr ***@***.***>
Sent: Friday, May 3, 2024 3:46 PM
To: TimMangan/MSIX-PackageSupportFramework ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TimMangan/MSIX-PackageSupportFramework] EnvVarFixup DLL Not Found (Discussion #25)


Hi
I downloaded the latest version of the MSIX-PackageSupportFramework project and I put the files into the root of my MSIX package. When I try to run the packaged MSIX program, it doesn't work. I ran the debug tools and found that PsfRuntime cannot find EnvVarFixup32.dll. I double checked my package and the EnvVarFixup32.dll is at the location PsfRuntime is looking at (see attached screenshots).

Any ideas on what I could try?

Thanks,
EmperorLrrr
EnvVarFixup.in.package.png (view on web)<https://github.com/TimMangan/MSIX-PackageSupportFramework/assets/95937528/9f5008e4-0c98-4fac-9fee-2388e8199db1>
debug.output.png (view on web)<https://github.com/TimMangan/MSIX-PackageSupportFramework/assets/95937528/1b0de3cc-9db4-4ab2-b42c-5659595dcfa2>

**Comment by EmperorLrrr** [comment]
Hi

Thank you for the quick response while travelling.   I was finally able to to circle back to this item and adding the libraries to the MSIX package fixed the EnvVarFixup DLL not being found.   Sadly I'm running into other issues with the application I'm trying to package, but that is another issue.  Probably due to it being a custom built program that is not "off the shelf" software.  It still has VB6 code in it.  I'm going to rebuild the MSIX from scratch just to rule out any weirdness with the current package.

Thanks for your help,
EmperorLrrr

**Comment by TimMangan** [comment]
VB6 Apps require write access, so adding MfrFixup in ILV mode and enabling ILV (InstalledLocationVIrtualization) in the AppXManifest would be the minimum.
________________________________
